### PR TITLE
CM-1568 Move the type usage to the types package

### PR DESCRIPTION
**Pull request by alexander2001i**
Moved types:

- [x] Common Types
- [x] Proposal Types
- [x] Vote Types
- [x] Subscription Types

-----
- [x] Fix the test files

### CODE REVIEW

File: packages/firebase/functions/src/discussion/database/getDiscussion.ts [hunk 1]
```diff
@@ -29,7 +28,7 @@ export const getDiscussion = async (discussionId: string, customOptions?: Partia
 
   const discussion = (await discussionCollection
     .doc(discussionId)
-    .get()).data() as Nullable<IDiscussionEntity | IProposalEntity>
+    .get()).data() as Nullable<IDiscussionEntity> // | IProposalEntity> @question(for: Moore) Why this can be proposal entity? Does not make any sense to me whatsoever
```
Comment (watchtower314) [inline packages/firebase/functions/src/discussion/database/getDiscussion.ts:32]: @alexander2001i because a discussion can also be connected to a proposal

Comment (alexander2001i) [reply packages/firebase/functions/src/discussion/database/getDiscussion.ts:32]: But here you are getting the discussion entity, not the related things

Comment (watchtower314) [reply packages/firebase/functions/src/discussion/database/getDiscussion.ts:32]: Here's another place we need to make the distinction:
Screen Shot 2021-02-03 at 6 57 11

Comment (watchtower314) [reply packages/firebase/functions/src/discussion/database/getDiscussion.ts:32]: Oh, I see what you're saying, one sec

Comment (watchtower314) [reply packages/firebase/functions/src/discussion/database/getDiscussion.ts:32]: Yes, you are right, it's not needed; I'm not sure why it seemed right at the time I wrote this 🤷🏽‍♀️

**Review by watchtower314 — APPROVED** [review]

**Pull request merged**